docs/mkDocs/docs/options.md

```diff
@@ -157,6 +157,54 @@ myIsDigit x = '0' <= x && x <= '9'
 {-@ assume reflect isDigit as myIsDigit @-}
 ```
 
+### Higher order reasoning
+
+To make PLE aware of eta-expansion and beta-reduction rules, you can use the
+`--etabeta` flag:
+
+```
+{-@ LIQUID "--etabeta" @-}
+```
+
+```Haskell
+{-@ eta :: f:_ -> { f = \x:Int -> f x } @-}
+eta :: (Int -> Int) -> Proof
+eta _ = trivial
+
+{-@ beta :: { (\x:Int -> x) (12) = 12 } @-}
+beta :: Proof
+beta = trivial
+```
+
+This also allows PLE to unfold partially applied reflected functions.
+
+Note: The eta-expansion rule subsumes the `--extensionality` flag. To save time
+during typechecking, you can disable `--extensionality` when using `--etabeta`.
```

Comment:
I checked if enabling --etabeta would have the effect of enabling extensionality, but this is not the case since there is [one place](https://github.com/ucsd-progsys/liquidhaskell/blob/d90769a9cbf7641881e8a62500a7a6fc0fa32aff/liquidhaskell-boot/src/Language/Haskell/Liquid/Constraint/Generate.hs#L506) in LH where the extensionality flag is used and etabeta is ignored.

I don't know what it is meant by the first sentence: "The eta-expansion rule subsumes the `--extensionality` flag".

> To save time￼during typechecking,

Do you mean that extensionality makes etabeta noticeably slower?

Reply:
But we can easily not mention this if you think is only going to confuse the users

Reply:
I made the note more explicit in e6dc9ac.

What do you think of [that use](https://github.com/ucsd-progsys/liquidhaskell/blob/d90769a9cbf7641881e8a62500a7a6fc0fa32aff/liquidhaskell-boot/src/Language/Haskell/Liquid/Constraint/Generate.hs#L506) of extensionality in LH?

Should that place take into account etabeta as well?

Reply:
I think that code is needed to take into account the fact that extensionality flag is changing the type of some expressions, thing that is not done by doing eta expansion, also the eta expansion is totally transparent to LH as it's actually done completely in LF so it shouldn't require any modification in LH